### PR TITLE
More flexible API

### PR DESCRIPTION
**Pull request by dubiousdavid**
```typescript
loadSnapshot(msg: JsMsg): Promise<Snapshot> | Snapshot
persistSnapshot(snapshot: StreamSnapshot, msg: JsMsg): void
```

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -35,20 +35,21 @@ export const runTopologyWithNats =
...
-    const streamData = getStreamDataFromMsg(msg)
     const isRedelivery = numAttempts > 1
+    debug('Redelivery %s', isRedelivery)
     const { emitter, promise, getSnapshot } = isRedelivery
       ? // Resume topology based on unique stream data
-        resumeTopology(spec, await loadSnapshot(streamData))
+        resumeTopology(spec, await loadSnapshot(msg))
       : // Run topoloty with data from msg
-        runTopology(spec, dag, { data, meta: streamData })
+        runTopology(spec, dag, { data })
```
Comment (stellarhoof) [inline src/index.ts:44]: One thing I noticed is that there's no way of passing `{include,exclude}Nodes` to `runTopology`

File: src/index.ts [hunk 2]
```diff
@@ -35,20 +35,21 @@ export const runTopologyWithNats =
...
+    const streamData = getStreamDataFromMsg(msg)
     const persist = (snapshot: Snapshot) => {
       const streamSnapshot = { ...snapshot, ...streamData, numAttempts }
```
Comment (stellarhoof) [inline src/index.ts:47]: So you decided to leave `streamData`?

Comment (dubiousdavid) [reply src/index.ts:47]: Yeah. It was annoying to have to add that data. It's useful information even if you don't use it as a uniqueness constraint.

File: src/index.ts (hunk 1 above)
Comment (dubiousdavid) [reply src/index.ts:44]: The thinking was that you could just use the exported fn `filterDAG` from `topology-runner`.

Comment (stellarhoof) [reply src/index.ts:44]: We should only have one way of doing things. Either `filterDAG` or `include`/`exclude` for both.

Comment (dubiousdavid) [reply src/index.ts:44]: Makes sense, but I think that can be addressed in a separate PR.

Comment (dubiousdavid) [reply src/index.ts:44]: I'd likely just remove it from `runTopology` if we want to keep things consistent.

Comment (stellarhoof) [reply src/index.ts:44]: Yeah I don't have any strong opinions. Something we should consider is that if we go with `filterDAG`, every job would still need to take `include/exclude` nodes, so we have the ability to publish a message to run a subset of the dag. E.x:

```
./pub.js federalSamOpportunity '({ params: { ... }, includeNodes: [...] })'
```

**Review by stellarhoof — APPROVED** [review]

**Pull request merged**